### ISSUE TITLE
Error: Failed to write to the YubiKey. Make sure the device does not have restricted access.

### ISSUE THREAD

**Issue by schiermi** [Yubico/yubikey-manager#368]
- YubiKey Manager (ykman) version:
3.1.1
- How was it installed?:
zypper install yubikey-manager
- Operating system and version:
OpenSuSE Tumbleweed
- YubiKey model and version:
YubiKey Standard 2.2.3
- Bug description summary:
Error: Failed to write to the YubiKey. Make sure the device does not have restricted access.

Steps to reproduce
---
ykman otp static -f 1 test

[Please explain what you did when the bug appeared, and if and how you have been
able to reproduce it.]
After successfully deleting both slots (therefore: device access is not restricted) I tried to configure them with new OTP configurations which fails. No matter which OTP Application (static, yubiotp…) I try.

Expected result
---

[What did you expect to happen when you did the above?]
Configuring my Yubikeys slot with the new configuration

Actual results and logs
---
2020-12-29T23:13:06+0100 INFO [ykman.logging_setup.setup:58] Initialized logging for ykman version: 3.1.1
2020-12-29T23:13:06+0100 DEBUG [ykman.descriptor.Descriptor.open_device:86] transports: 0x1, self.mode.transports: 0x1
2020-12-29T23:13:06+0100 DEBUG [ykman.descriptor.open_device:91] Opening driver for serial: None, type: YUBIKEY.YKS, mode: OTP
2020-12-29T23:13:06+0100 DEBUG [ykman.descriptor.open_device:94] Attempt 1 of 10
2020-12-29T23:13:06+0100 DEBUG [ykman.driver_otp.open_devices:224] Success in opening key at position 0
2020-12-29T23:13:06+0100 DEBUG [ykman.descriptor.open_device:97] Found driver: <ykman.driver_otp.OTPDriver object at 0x7f5b09a54430>, key_type: YUBIKEY.YKS, mode: OTP
2020-12-29T23:13:06+0100 DEBUG [ykman.device.__init__:198] Read config from device...
2020-12-29T23:13:06+0100 DEBUG [ykman.device.__init__:213] Failed to read config from device
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/ykman/device.py", line 199, in __init__
    config = DeviceConfig(driver.read_config())
  File "/usr/lib/python3.8/site-packages/ykman/driver_otp.py", line 137, in read_config
    raise NotSupportedError()
ykman.driver.NotSupportedError
2020-12-29T23:13:06+0100 DEBUG [ykman.device.__init__:119] Config data empty/missing
2020-12-29T23:13:06+0100 DEBUG [ykman.device.__init__:247] YK Standard identified
2020-12-29T23:13:06+0100 ERROR [ykman.cli.otp._failed_to_write_msg:77] Failed to write to device
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/ykman/cli/otp.py", line 397, in static
    controller.program_static(slot, password, keyboard_layout, SlotConfig(
  File "/usr/lib/python3.8/site-packages/ykman/otp.py", line 288, in program_static
    with self._create_cfg(cmd, config) as cfg:
  File "/usr/lib/python3.8/site-packages/ykman/otp.py", line 192, in _create_cfg
    context = _SlotConfigContext(self._dev, cmd, conf)
  File "/usr/lib/python3.8/site-packages/ykman/otp.py", line 147, in __init__
    self._apply(conf)
  File "/usr/lib/python3.8/site-packages/ykman/otp.py", line 153, in _apply
    check(ykpers.ykp_set_extflag(self.cfg, 'ALLOW_UPDATE'))
  File "/usr/lib/python3.8/site-packages/ykman/driver_otp.py", line 71, in check
    raise YkpersError(ykpers.yk_get_errno())
ykman.driver_otp.YkpersError: ykpers error 0, b''
2020-12-29T23:13:06+0100 DEBUG [ykman.driver_otp.close:203] Close <ykman.driver_otp.OTPDriver object at 0x7f5b09a54430>
Usage: ykman otp static [OPTIONS] [1|2] [PASSWORD]
Try 'ykman otp static -h' for help.

Error: Failed to write to the YubiKey. Make sure the device does not have restricted access.
2020-12-29T23:13:06+0100 DEBUG [ykman.driver_otp.__del__:208] Destroy <ykman.driver_otp.OTPDriver object at 0x7f5b09a54430>

[What actually happened?]

Other info
---

[Anything else you would like to add?]

**Comment by dainnilsson** [comment]
Please try updating to ykman 4.0 to see if that helps. If it doesn't, please provide new logs using 4.0 and reopen this issue.

**Comment by schiermi** [comment]
One week ago I put the key to my electronic scraps for disposal.
Recovered it, tried with 4.0.0 - works. Thanks a lot!

**Comment by ares487** [comment]
I'm not sure if this is your issue but if the problem is the "ERROR: Failed to write to the YubiKey. Make sure the device does not have restricted access (see "ykman otp --help" for more info)." code, and you are not 100% sure there is an access code on the slot from a previous user (e.g. if the key was bought new and was fresh out of tear-to-open packaging), then I think I have the solution.

I've just been getting railed for a couple hours by this issue and I fixed it by opening the windows version of yubico authenticator ver7.4.0 at https://www.yubico.com/products/yubico-authenticator/#h-download-yubico-authenticator as of 6th July 2026 (guessing any desktop version is fine just not mobile because I don't see the slots menu there), going to the slots menu on the left hand side, selecting the slot I wanted to configure, choosing the setup option (in my case challenge-response), generating a random key (to switch the slot configuration to that security protocol), then re-attempting configuration where I originally got violated and viola, it's alive! My intention was to make both slots challenge-response but the stock config of slot 1 is OTP so when I tried to select slot 1 to configure with challenge-response, I got the aforementioned error. Hope this helps
